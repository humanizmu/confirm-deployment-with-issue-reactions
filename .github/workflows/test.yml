name: 'build-test'
on: # rebuild any PRs and main branch changes
  push:

jobs:
  build: # make sure build/ci work properly
    timeout-minutes: 2
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - run: |
          npm install
      - run: |
          npm run all
  test: # make sure the action works on a clean machine without building
    timeout-minutes: 2
    runs-on: ubuntu-latest
    # if: github.ref == 'refs/heads/main'
    steps:
      - uses: actions/checkout@v2
      - uses: ./
        # with:
        #   githubToken: ${{ secrets.GITHUB_TOKEN }}
