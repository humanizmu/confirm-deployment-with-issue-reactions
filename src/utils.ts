import * as core from '@actions/core'

export const logConfirmationIssueUrl = (issueUrl: string) => {
  core.info(`Opened issue ${issueUrl}`)
  core.info(`Confirm this step by reacting to the issue with 👍`)
  core.info('To cancel this step react to the issue with 👎')
}
