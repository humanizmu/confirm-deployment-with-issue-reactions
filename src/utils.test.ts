import * as core from '@actions/core'
import {castToMock} from './test/type-utils'
import {logConfirmationIssueUrl} from './utils'

jest.mock('@actions/core')
const coreInfoMock = castToMock(core.info)

describe('utils', () => {
  describe('logConfirmationIssueUrl()', () => {
    beforeEach(() => {
      logConfirmationIssueUrl('http://putin.huilo')
    })

    it('logs issue url and explanation messages correctly', () => {
      expect(coreInfoMock.mock.calls).toMatchSnapshot()
    })
  })
})
