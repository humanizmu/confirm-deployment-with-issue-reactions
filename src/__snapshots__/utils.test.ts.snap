// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils logConfirmationIssueUrl() logs issue url and explanation messages correctly 1`] = `
Array [
  Array [
    "Opened issue http://putin.huilo",
  ],
  Array [
    "Confirm this step by reacting to the issue with 👍",
  ],
  Array [
    "To cancel this step react to the issue with 👎",
  ],
]
`;
